DOCUMENTATION.md

```diff
@@ -0,0 +1,196 @@
+
+if you want to rebuild some targets chain each time please see How geocint pipeline workyou run the pipeline, you must add the initial target of this chain to the TARGETS_TO_CLEAN variable in the configuration file config.inc.sh).
+For example, you have a target’s chain like a (here is a simple chain of targets, just target’s name):
+```
+data/in/some.tiff: | data/in ## initial target, download input data
+data/mid/data_tiff.csv: data/in/some.tiff | data/mid## extract data from raster to intermediate file
+db/table/table_from_csv: data/mid/data_tiff.csv | db/table ## load data from CSV to database
+data/out/output.geojson.gz: db/table/table_from_csv | data/out ## extract data from database to output file
+```
+If you want to rebuild this chain of targets each time when you run the pipeline, you should add data/in/some.tiff target to the TARGETS_TO_CLEAN variable. It means that before each run make will remove data/in/some.tiff if exists and rebuild them and all targets, that have it in list of dependencies (will check dependencies recursively).
+If you do not want to rebuild this chain, just make sure that the final target from your chain doesn’t depend on targets from the TARGETS_TO_CLEAN variable. This method can be useful when you have a large dataset that has not been updated recently and requires heavy and time-consuming pre-processing. In such a case, you probably want to prepare the data and load it into the database once and not repeat this work without changing the input data.
+But! Keep in mind, that dependencies inherited recursively, even if you don’t put data/in/some.tiff in the data/out/output.geojson.gz dependencies list, them will be data/out/output.geojson.gz’s dependency. This means that these two lines are equivalent:
+```
+data/out/output.geojson.gz: db/table/table_from_csv
+data/out/output.geojson.gz: data/in/some.tiff db/table/table_from_csv db/table/table_from_csv | data/out
+```
+You can read more about target's dependencies(prerequisites) in the official [GNU Make manual](https://www.gnu.org/software/make/manual/make.html#Prerequisite-Types)
+
+### User schemas
+
+User schemas can be used for separate pipeline and dev data.
+Run [scripts/create_geocint_user.sh](scripts/create_geocint_user.sh) to initialize the user schema.
+
+`sudo scripts/create_geocint_user.sh [username]`
+
+Script for adding user role and schema to geocint database. If no username is provided, it will be prompted. User roles
+are added to the geocint_users group role. You need to add the following line to pg_hba.conf.
+
+`local   gis +geocint_users  trust`
+
+### How to analyse build time for tables
+
+Logs for every build are stored in `/home/gis/geocint/logs`
+
+This command can show lastN {*Total times in ms*} for some {*tablename*} ordered by date
+
+```bash
+cd /home/gis/geocint/logs
```

Comment:
this can be deleted and added as one command in find

Reply:
replaced